tests/networks.spec.ts

```diff
@@ -4,148 +4,370 @@ import {
 
-test.beforeAll(async ({ browser, browserName }) => {
-  // network names can only be 15 characters long
-  network = `${browserName.substring(0, 2)}-${network}`;
-  const page = await browser.newPage();
-  await createNetwork(page, network);
-  await page.close();
-});
+test.describe("bridge type", () => {
+  test.beforeAll(async ({ browser, browserName }) => {
+    // network names can only be 15 characters long
+    network = `${browserName.substring(0, 2)}-${network}`;
+    const page = await browser.newPage();
+    await createNetwork(page, network);
+    await page.close();
+  });
+
+  test.afterAll(async ({ browser }) => {
+    const page = await browser.newPage();
+    await deleteNetwork(page, network);
+    await page.close();
+  });
+
+  test("configure main bridge network settings", async ({ page }) => {
+    const options = ["false", "true"];
+
+    await visitNetwork(page, network);
+    await editNetwork(page);
+
+    await page
+      .getByPlaceholder("Enter description")
+      .fill(NETWORK_CONSTANTS.DESCRIPTION);
+
+    const radioButtons = await page.$$("input[type=radio]");
+    for (const button of radioButtons) {
+      await button.click({ force: true });
+    }
+
+    await page.getByLabel("IPv4 address").fill(NETWORK_CONSTANTS.IPV4_ADDRESS);
+    await page.getByLabel("IPv6 address").fill(NETWORK_CONSTANTS.IPV6_ADDRESS);
+
+    for (const option of options) {
+      await setOption(page, "IPv4 NAT", option);
+      await setOption(page, "IPv6 NAT", option);
+    }
+  });
+
+  test("set bridge network driver and hardware details", async ({ page }) => {
+    const options = ["openvswitch", "native"];
+
+    await prepareNetworkTabEdit(page, "Bridge", network);
+
+    await setTextarea(page, "MTU", NETWORK_CONSTANTS.MTU);
+
+    await setTextarea(
+      page,
+      "Hardware address",
+      NETWORK_CONSTANTS.HARDWARE_ADDRESS,
+    );
+
+    for (const option of options) {
+      await setOption(page, "Bridge driver", option);
+    }
+  });
+
+  test("configure bridge DNS settings", async ({ page }) => {
+    const options = ["none", "managed", "dynamic"];
+
+    await prepareNetworkTabEdit(page, "DNS", network);
+
+    await setTextarea(page, "DNS domain", NETWORK_CONSTANTS.DNS_DOMAIN);
+    for (const option of options) {
+      await setOption(page, "DNS mode", option);
+    }
+    await setTextarea(page, "DNS search", NETWORK_CONSTANTS.DNS_SEARCH);
+  });
 
-test.afterAll(async ({ browser }) => {
-  const page = await browser.newPage();
-  await deleteNetwork(page, network);
-  await page.close();
+  test("set bridge IPv4 DHCP and range configuration", async ({ page }) => {
+    const options = ["false", "true"];
+    const DHCP_RANGES = "192.168.1.100-192.168.1.150";
+    const OVN_RANGES = "10.141.1.50-10.141.1.99";
+    const DHCP_EXPIRY = "2h";
+
+    await prepareNetworkTabEdit(page, "IPv4", network);
+
+    await page
+      .getByRole("row")
+      .filter({
+        hasText: "Whether to allocate",
```

Comment:
This text is missing when testing against the LXD 5.0 backend. Either use a different text for it like "IPv4 DHCPtrue" or skip this bit of the test for the 5.0 backend. See other tests for how to identify the backend version from test code.